src/terrain/Terrain.h

```diff
@@ -43,7 +43,7 @@ class Terrain : public RefCounted {
 		return m_fracdef[index];
 	}
 
-	virtual double GetHeight(const vector3d &p) const = 0;
+	virtual void GetHeights(const std::vector<vector3d> &vP, std::vector<double> &heightsOut) const = 0;
```

Comment:
Note: in lieu of a proper span type, I'd recommend this function signature instead be:
```suggestion
	virtual void GetHeights(const vector3d *vP, double *heightsOut, size_t count) const = 0;
```

This prevents the need to construct a single-element vector, properly expresses that the number of vertices must equal the number of heights generated, and allows for more flexibility in the underlying storage.

Regardless of function signature, bounds checking *must* be performed in the calling function - this just makes that more explicit.

ALSO: you *really* don't want to be calling the bounds-checking `std::vector::at` method for every single iteration of the noise function, nor do you want to be resizing the vector mid-fill. Both have significant performance implications at the quantity of vertices we process.